src/bot.ts

```diff
@@ -556,20 +556,26 @@ async function withRetries<T>(
     const INITIAL_DELAY = 100; // ms
     let delay = INITIAL_DELAY;
     while (!result.ok) {
-        let mustDelay = true;
+        let mustDelay = false;
         try {
             result = { ok: true, value: await task() };
-            mustDelay = false;
         } catch (error) {
             debugErr(error);
-            if (error instanceof HttpError) continue;
+            if (error instanceof HttpError) {
+                mustDelay = true;
+                continue;
```

Comment:
mustDelay is set to true then the `continue` jumps to the next while loop iteration and directly sets it to false in line 559. Something is fishy here.

Reply:
The `finally` block takes care of this. It's not the prettiest code I've written, but it should work, eh?

Reply:
Yes, it's not very nice. It makes sense when you think about it, but it does not make sense until you do. Do you approve anyway?